### PR TITLE
RavenDB-7314 allowing to use starts with filters for smuggler

### PR DESCRIPTION
**Pull request by talweiss1982**

### CODE REVIEW

**Comment by ayende** [comment]
Why not use https://github.com/ravendb/ravendb/blob/v3.5/Raven.Database/Util/WildcardMatcher.cs ?

**Comment by talweiss1982** [comment]
I was not aware of it, it is just one line of code not sure if this is worth changing.

**Comment by ayende** [comment]
It is hard to follow the code, in the sense that there might be NRE or out of range there.

**Comment by talweiss1982** [comment]
that is why i test my code before sending PR, i'll change it...

**Comment by talweiss1982** [comment]
done
